crates/solver/src/settlement_rater.rs

```diff
@@ -362,10 +358,15 @@ impl ScoreCalculator {
             self.score_cap.clone(),
         )?;
         if optimal_score > *objective_value {
-            return Err(ScoringError::ScoreHigherThanObjective(
-                optimal_score,
-                objective_value.clone(),
-            ));
+            tracing::error!(
+                "Sanity check failed, score higher than objective, should never happen unless a \
+                 bug in a computation"
```

Comment:
Nit: "unless _there's_ a bug". (Same in line below.)
Maybe it makes sense to also print the values? If we see that they differ by a rounding error we could start looking in different places compared to seeing two completely different values.

Reply:
There are few enough arguments going into this function that I think we could simply log all of them in the error. That way we wouldn't have to figure out how to get the relevant data in case we ever encounter this error.

Reply:
Fixed comment and added more parameters to log.